core/src/main/java/org/apache/iceberg/hadoop/SerializedConfiguration.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hadoop;
+
+import java.io.Serializable;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+/**
+ * A class that captures the current state of a Hadoop configuration in a serializable manner.
+ * <p>
+ * <em>Note:</em> Unlike {@link SerializableConfiguration}, no subsequent changes to the underlying
+ * Hadoop configuration will be propagated to the captured state once an instance
+ * of {@link SerializedConfiguration} is constructed.
+ */
+public class SerializedConfiguration implements Serializable {
+
+  private final Map<String, String> confAsMap;
+  private transient volatile Configuration conf = null;
+
+  public SerializedConfiguration(Configuration conf) {
+    this.confAsMap = Maps.newHashMapWithExpectedSize(conf.size());
+    conf.forEach(entry -> confAsMap.put(entry.getKey(), entry.getValue()));
```

Comment:
I think this should also set `conf` to the one that is passed in. That way this won't create a new configuration if `get` is called before the object is serialized.

Reply:
If we do this, there will be no guarantee the conf and map are in sync. This class should be used right before the serialization and should not be accessed before it is serialized.

Reply:
I moved this class to `SerializedTable` to hide it and make sure nobody uses it.